buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -147,6 +150,38 @@ class ClusterFormationTasks {
                     into node.baseDir
                 }
                 break;
+            case 'rpm':
+                File rpmDatabase = new File(node.baseDir, 'rpm-database')
+                File rpmExtracted = new File(node.baseDir, 'rpm-extracted')
+                /* Delay reading the location of the rpm file until task execution */
+                Object rpm = "${ -> project.configurations.elasticsearchDistro.singleFile}"
+                extract = project.tasks.create(name: name, type: Exec, dependsOn: extractDependsOn) {
+                    commandLine 'rpm', '--badreloc', '--nodeps', '--noscripts', '--notriggers',
+                        '--dbpath', rpmDatabase,
+                        '--relocate', "/=${rpmExtracted}",
+                        '-i', rpm
+                    standardOutput = new ByteArrayOutputStream()
+                    errorOutput = standardOutput
+                    /* rpm complains about the database being corrupted and exits. But it gets far
+                      enough for us to use it as an extractor. This is kind of funky but it works
+                      and its how Elasticsearch's maven build used to do it. */
+                    ignoreExitValue true
+                    doLast {
+                        String out = standardOutput.toString()
+                        if (out.indexOf('DB_RUNRECOVERY') < 0) {
```

Comment:
On OSX for me but it looks like its not causing the process to exit-non-0. So I'll change this to LoggedExec.